src/Time/Units.hs

```diff
@@ -136,6 +136,18 @@ instance KnownSymbol (ShowUnit unit) => Read (Time unit) where
             unless (timeUnitStr == symbolVal (Proxy @(ShowUnit unit))) pfail
             pure $ Time (n % m)
 
+instance Num (Time unit) where
+    (Time t1) + (Time t2) = Time (t1 + t2)
```

Comment:
Maybe you can add `{-# INLINE #-}` pragmas. Also you can implement all methods (except `error` ones) by delegating to `coerce` function how it's done here:

* https://hackage.haskell.org/package/base-4.10.1.0/docs/src/Data.Semigroup.html#line-243

So, this should work I guess.

```haskell
(+) = coerce ((+) :: RatioNat -> RatioNat -> RatioNat)
```